flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/JobExceptionsHandler.java

```diff
@@ -71,7 +72,11 @@ public JobExceptionsHandler(
 	}
 
 	@Override
-	protected JobExceptionsInfo handleRequest(HandlerRequest<EmptyRequestBody, JobMessageParameters> request, AccessExecutionGraph executionGraph) {
+	protected JobExceptionsInfo handleRequest(HandlerRequest<EmptyRequestBody, JobExceptionsMessageParameters> request, AccessExecutionGraph executionGraph) {
+		List<Integer> sizes = request.getQueryParameter(ExceptionShowSizeParameter.class);
+		if (sizes != null && sizes.size() == 1) {
```

Comment:
the returned list is never null. The size check is overly strict and should just check `sizes.size() > 0`. It is fine to use the first value should the parameter be defined multiple times.

Reply:
As MessageQueryParameter's type is List<X>, sizes  has to define to List<Integer>. It's no good. Maybe we could split MessageQueryParameter to MessageQueryListParameter and MessageQueryObjectParameter?

Reply:
The fact of the matter is that users can provide multiple value for query parameters. Hence, you have to handle this case somehow.

Again, just use the first value.